### ISSUE TITLE
Question about loss function

### ISSUE THREAD

**Issue by yangyingni** [cs-giung/FSRNet-pytorch#5]
Sorry to bother you so late.I have a question about loss function.
Question 1: I saw your overview
- Train Coarse SR Network to minimize (1).
- Train Coarse SR Network, and Prior Estimation Network to minimize (2).
- Train whole network to minimize (3).
Shouldn't it be Train whole network to minimize (1)+(2)+(3)?

**Comment by cs-giung** [comment]
In FSRNet, whole network is trained by minimizing (1)+(2)+(3). See:
https://github.com/cs-giung/FSRNet-pytorch/blob/master/train.py#L87

Note that the words in the **Overview** only point to sub-networks that affect each loss.
(e.g. `(1) MSE Loss between Coarse SR Image and Ground Truth` relates only to `Coarse SR Network`, since `Coarse SR Image` is obtained through only `Coarse SR Network`)

**Comment by cs-giung** [comment]
I revised README:
https://github.com/cs-giung/FSRNet-pytorch/commit/5dd5f2d115937dd0c1740e41d7058a66b99cb8fe

**Comment by yangyingni** [comment]
Thank you for your answer.You mean this model is trained once with loss=loss1+loss2+loss3?Should i pretrain estimation network firstly,if need pretrain,how to load parameter to this model?





------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"notifications"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月26日(星期四) 晚上10:37
收件人:&nbsp;"cs-giung/FSRNet-pytorch"<FSRNet-pytorch@noreply.github.com&gt;;
抄送:&nbsp;"泱渶"<2676426668@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [cs-giung/FSRNet-pytorch] Question about loss function (#5)





 
In FSRNet, whole network is trained by minimizing (1)+(2)+(3). See:
 https://github.com/cs-giung/FSRNet-pytorch/blob/master/train.py#L87
 
Note that the words in the Overview only point to sub-networks that affect each loss.
 (e.g. (1) MSE Loss between Coarse SR Image and Ground Truth relates only to Coarse SR Network, since Coarse SR Image is obtained through only Coarse SR Network)

**Comment by cs-giung** [comment]
The authors trained FSRNet from scratch.

Perhaps because FSRNet is an independent structure from the existing ImageNet models (e.g. VGG, ResNet).

**Comment by yangyingni** [comment]
Thank you very much.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"notifications"<notifications@github.com&gt;;
发送时间:&nbsp;2020年3月26日(星期四) 晚上10:54
收件人:&nbsp;"cs-giung/FSRNet-pytorch"<FSRNet-pytorch@noreply.github.com&gt;;
抄送:&nbsp;"泱渶"<2676426668@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [cs-giung/FSRNet-pytorch] Question about loss function (#5)





 
The authors trained FSRNet from scratch.
 
Perhaps because FSRNet is an independent structure from the existing ImageNet models (e.g. VGG, ResNet).
